src/ir/assigments.rs

```diff
@@ -40,6 +40,19 @@ impl<F: Clone, TraceArgs> Clone for AssigmentGenerator<F, TraceArgs> {
     }
 }
 
+impl<F: Clone, TraceArgs> Default for AssigmentGenerator<F, TraceArgs> {
+    fn default() -> Self {
+        Self {
+            columns: Default::default(),
+            placement: Default::default(),
+            selector: Default::default(),
+            trace_gen: Default::default(),
+            num_rows: Default::default(),
+            ir_id: Default::default(),
+        }
+    }
+}
```

Comment:
All use default. Can be derived then?

Reply:
To be derived it requires F and TraceArgs to be Default, which is not really necessary.